Apsitvarkom.Utilities/Logging/FileLogger.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Apsitvarkom.Configuration;
+using Microsoft.Extensions.Logging;
+using System.Collections.Concurrent;
+
+namespace Apsitvarkom.Utilities;
+
+public class FileLogger : ILogger, IDisposable
+{
+    private readonly FileLoggerConfiguration _configuration;
+    private readonly BlockingCollection<string> messagesQueue = new (new ConcurrentQueue<string>());
+    private readonly CancellationTokenSource _cancellationTokenSource;
+    private readonly CancellationToken _cancellationToken;
+
+    /// <summary>Constructor that is being called by <see cref="FileLoggerProvider"/>.</summary>
+    public FileLogger(FileLoggerConfiguration configuration)
+    {
+        _configuration = configuration;
+
+        // Start a new thread that is solely responsible for writing to a file for this logger
+        _cancellationTokenSource = new CancellationTokenSource();
+        _cancellationToken = _cancellationTokenSource.Token;
+        Task.Run(() => WriteLogToFile(), _cancellationToken);
```

Comment:
```suggestion
        Task.Run(WriteLogToFile, _cancellationToken);
```

not necessarily a needed change, but this is possible too :D

Reply:
Resharper's suggestion btw

Reply:
yeah makes sense